pkg/cmd/bazci/bazci.go

```diff
@@ -499,24 +412,33 @@ func removeEmergencyBallasts() {
 	}
 }
 
-func processTestJSONIfNeeded(shouldCreateTarball bool, goTestJSONOutputFilePath string) error {
-	if !shouldProcessTestFailures {
-		return nil
-	}
+func processTestXmls(testXmls []string) error {
 	removeEmergencyBallasts()
-	buf, err := os.ReadFile(goTestJSONOutputFilePath)
-	if err != nil {
-		return err
-	}
-	if err := processFailures(buf, filepath.Join(artifactsDir, "failures.txt")); err != nil {
-		return err
-	}
-	if err := postReleaseOnlyFailures(buf); err != nil {
-		return err
-	}
-	if shouldCreateTarball {
-		if err := createTarball(goTestJSONOutputFilePath); err != nil {
-			return err
+	branch := strings.TrimPrefix(os.Getenv("TC_BUILD_BRANCH"), "refs/heads/")
+	isReleaseBranch := strings.HasPrefix(branch, "master") || strings.HasPrefix(branch, "release") || strings.HasPrefix(branch, "provisional")
+	if isReleaseBranch {
+		// GITHUB_API_TOKEN must be in the env or github-post will barf if it's
+		// ever asked to post, so enforce that on all runs.
+		// The way this env var is made available here is quite tricky. The build
+		// calling this method is usually a build that is invoked from PRs, so it
+		// can't have secrets available to it (for the PR could modify
+		// build/teamcity-* to leak the secret). Instead, we provide the secrets
+		// to a higher-level job (Publish Bleeding Edge) and use TeamCity magic to
+		// pass that env var through when it's there. This means we won't have the
+		// env var on PR builds, but we'll have it for builds that are triggered
+		// from the release branches.
+		if os.Getenv("GITHUB_API_TOKEN") == "" {
+			return errors.New("GITHUB_API_TOKEN must be set")
+		}
+		for _, testXml := range testXmls {
+			xmlFile, err := os.Open(testXml)
+			if err != nil {
+				return err
+			}
+			githubpost.PostFromTestXML(githubPostFormatterName, xmlFile)
+			if err := xmlFile.Close(); err != nil {
+				return err
+			}
 		}
```

Comment:
```suggestion
    postErrors := make([]string, 0, len(testXmls))
		for _, testXml := range testXmls {
			xmlFile, err := os.Open(testXml)
			if err != nil {
			  postErrors = append(postErrors, fmt.Sprintf("Failed to open %s with the following error: %v", testXml, err))
				continue
			}
			if err := githubpost.PostFromTestXML(githubPostFormatterName, xmlFile); err != nil {
			  postErrors = append(postErrors, fmt.Sprintf("Failed to process %s with the following error: %v", testXml, err))
			continue
			}
			if err := xmlFile.Close(); err != nil {
			  fmt.Printf("Failed to close %s with error: %v\n", testXml, err)
			}
		}
		if len(postErrors) != 0 {
		    return errors.New(strings.Join(postErrors, "\n"))
		}
```

Reply:
should we try to process as much files as possible instead of skipping remaining `test.xml` files as soon as an error occurs?

Reply:
Don't see why not; it would be an improvement over the original code (in `listFailures`) which bailed out on the first parsing or posting error.